### DISCUSSION TITLE
Online Presence

### DISCUSSION THREAD

**Discussion by mathiascode** [nicotine-plus/nicotine-plus#1594] [General]
Matters regarding the online presence of Nicotine+.

**Comment by mathiascode** [comment]
SourceForge project page is very outdated
https://github.com/nicotine-plus/nicotine-plus/issues/1477

**Comment by mathiascode** [comment]
GNU Guix project page is outdated
https://github.com/nicotine-plus/nicotine-plus/issues/1478

**Reply by mathiascode** [reply]
Looks like someone has picked up the package again: https://issues.guix.gnu.org/54273

**Comment by mathiascode** [comment]
Open Source Libs project page is outdated
https://github.com/nicotine-plus/nicotine-plus/issues/1479

**Reply by mathiascode** [reply]
Seems to be fixed now.

**Comment by mathiascode** [comment]
Edit: done

Point nicotine-plus.org domain directly to GitHub Pages
https://github.com/nicotine-plus/nicotine-plus/issues/409
CC @eLvErDe

**Comment by slook** [comment]
Old version of Nicotine+ 1.2.16 on Free Download Manager (spammy aggregate site "Latest update: Nov 15, 2021"):

https://en.freedownloadmanager.org/Windows-PC/Nicotine-FREE.html

**Comment by slook** [comment]
**Link to Release notes on Downloads page doesn't work properly**

Download the current stable version of Nicotine+ for your operating system. For the release notes, see [NEWS.md](https://nicotine-plus.org/NEWS.md).

It downloads the .md file instead of going to the web page.

https://github.com/nicotine-plus/nicotine-plus/blob/5ef15cd9a146d3410abd95b4921a4336168819f8/doc/DOWNLOADS.md?plain=1#L3

**Reply by mathiascode** [reply]
Huh, wonder when this started occurring.

Edit: apparently it was always like that, according to wayback machine http://web.archive.org/web/20220310044858/https://nicotine-plus.org/doc/DOWNLOADS.html

**Reply by mathiascode** [reply]
Fixed in 2bed21ad8ef87712ac98bd77cdc4d48c5880e41f

**Comment by slook** [comment]
Article exists on on Wikipedia but it is not translated into English:

https://nl.wikipedia.org/wiki/Nicotine%2B "Nicotine+" - full article about plus references version 3.3.0 but with outdated 1.2.4 screenshot.

https://nl.wikipedia.org/wiki/Nicotine_(software) "Nicotine (software)" - full article about non-plus version.

"Nicotine (software)" also has no English version, instead it does redirect to the Soulseek page, link:

https://en.wikipedia.org/w/index.php?title=Nicotine_(software)&redirect=no

https://www.wikidata.org/wiki/Q13895877 "Nicotine+" "client software for the Soulseek peer-to-peer file sharing network"
https://www.wikidata.org/wiki/Q2473191 "Nicotine (software)" - "abandoned client software for the SoulSeek peer-to-peer system"...

... where the old version says abandoned, it could do with reference to new "plus" version article.

**Comment by slook** [comment]
**Broken link in image attribution on WikiMedia**

![Nicotine-1](https://github.com/nicotine-plus/nicotine-plus/assets/88614182/44d1a6be-485e-4a3d-839a-66c08c4c9862)
_By Hyriand - http://nicotine.thegraveyard.org/nicotine-1.png, GPL, https://commons.wikimedia.org/w/index.php?curid=15069348_

(Broken link, that website is no longer up).

https://www.wikidata.org/wiki/Q2473191#/media/File:Nicotine-1.png

_FOUND_ - investigate to see if it's possible to add an alternative link to Archive.org link:
https://web.archive.org/web/20070928223725/http://nicotine.thegraveyard.org/nicotine-1.png

----

https://web.archive.org/web/20080515225005/http://nicotine.thegraveyard.org/

![image](https://github.com/user-attachments/assets/a0dacedc-7181-4c57-a3d7-0951396e1292)

![image](https://github.com/user-attachments/assets/8da429e4-ea40-4533-ac6e-8124a154de61)


----
https://web.archive.org/web/20110622022859/http://sensi.org/~ak/pyslsk/pyslsk1.png
![image](https://github.com/user-attachments/assets/0adb1202-c010-4b90-acea-c0634d871251)

https://web.archive.org/web/20110622023734/http://sensi.org/~ak/pyslsk/pyslsk2.png
![image](https://github.com/user-attachments/assets/0cce9b6a-2140-4f67-8192-02fe30456eeb)

https://web.archive.org/web/20110622022706/http://sensi.org/~ak/pyslsk/pyslsk3.png
https://web.archive.org/web/20110622023620/http://sensi.org/~ak/pyslsk/pyslsk4.png
https://web.archive.org/web/20110622023620/http://sensi.org/~ak/pyslsk/pyslsk5.png
![image](https://github.com/user-attachments/assets/86d2e2e8-18cd-4583-8b70-ee664044b456)

https://web.archive.org/web/20110622023053/http://sensi.org/~ak/pyslsk/pyslsk6.png
![image](https://github.com/user-attachments/assets/4166a2b6-5952-4a88-8cdf-218558e3c0d3)

----

https://web.archive.org/web/20080229070948/http://museek.thegraveyard.org/
![image](https://github.com/user-attachments/assets/14069a21-1ab4-461b-a566-84d695a8f1c1)

https://web.archive.org/web/20070717050700/http://museek.thegraveyard.org/museeq-20040320.png
![image](https://github.com/user-attachments/assets/56b73252-d4e1-4a47-85fa-af4add06def3)

https://web.archive.org/web/20070717050611/http://museek.thegraveyard.org/museeq-20040425-pm.png
![image](https://github.com/user-attachments/assets/8175de54-277d-43ee-944d-c3512e0a2ff1)

https://web.archive.org/web/20070717050740/http://museek.thegraveyard.org/museeq-20040425-search.png
![image](https://github.com/user-attachments/assets/a52cc376-a63e-4647-ab77-8eef63feae74)

https://web.archive.org/web/20070717050546/http://museek.thegraveyard.org/museeq-20040505.png
https://web.archive.org/web/20070717050755/http://museek.thegraveyard.org/museeq-20040505-2.png
https://web.archive.org/web/20070717050602/http://museek.thegraveyard.org/museeq-20040505-3.png

----

TODO: add screenshots folder with image assets in https://github.com/mathiascode/soulseek-archive

**Comment by slook** [comment]
_RESOLVED_ - **False meta-data description generated by GitHub releases page**

It appears on DuckDuckGo SERP page 1 for query "Nicotine+"...

![image](https://github.com/user-attachments/assets/4000a1c1-4ae4-4b85-af57-3d6128173b1a)

...Yet nowhere in the generated source is this kind of data provided. It must be some kind of hallucination which needs SEOing out.

**Reply by mathiascode** [reply]
Seems like DuckDuckGo is using ChatGPT to generate some of its result descriptions now? How embarrassing.

**Reply by slook** [reply]
Those hallucinated data are more likely to be coming from the `github.com` domain, since that is where many of the OpenAI (by Microsoft) models are being developed and have various sketchy implementations in CoPilot and whatnot. Alternatively, it may come from G or Bing, since DDG is not really an engine but is more like a search aggregator.

I don't know where one would need to go to make AI training data submissions for website SEO purposes, since it's not something that interests me much, and it is not directly relevant to the project.

There must be some tag or body content missing that has caused the crawler to fall back on similarity matching with other P2P softwares, as it seems to be making an assumption based on the most well known sharing protocol, without considering that there are in fact other less common ones that do exist.

**Reply by slook** [reply]
> DuckDuckGo AI Chat BETA
> You are chatting with Llama 3.1. AI chats may display inaccurate or offensive information.

>> what P2P protocol does the nicotine-plus project use?

> The Nicotine+ project uses the Soulseek protocol, which is a peer-to-peer (P2P) file sharing protocol.


>> What other P2P protocols exist?

> There are several other P2P protocols that exist, including:
> 
> • BitTorrent: a widely used protocol for distributing large files over the internet.
> • Gnutella: a decentralized P2P protocol that allows users to share files.
> • eDonkey: a P2P protocol that allows users to share files, particularly popular for sharing large files.
> • Kad: a decentralized P2P protocol that allows users to share files and is used by the eMule client.
> • OpenNap: a P2P protocol that allows users to share files, particularly popular in the early 2000s.
> • Ares: a P2P protocol that allows users to share files and communicate with each other.
> • Direct Connect: a P2P protocol that allows users to share files and communicate with each other.
> • RetroShare: a decentralized P2P protocol that allows users to share files, communicate, and create social networks.
> • I2P (Invisible Internet Project): a decentralized P2P protocol that allows users to share files and communicate anonymously.
> • Freenet: a decentralized P2P protocol that allows users to share files and communicate anonymously.
> 
> Note that this is not an exhaustive list, and there are many other P2P protocols that exist.
> 

>> How does GitHub generate the descriptions that appear on SERPs for projects like nicotine-plus?

> GitHub generates the descriptions that appear on Search Engine Results Pages (SERPs) for projects like Nicotine+ by using the project's README file. The README file is a text file that contains information about the project, such as its description, usage, and installation instructions.
> 
> When a project is indexed by a search engine like Google, the search engine's crawler will typically look for a meta description tag in the project's HTML page. However, since GitHub projects are hosted on GitHub's servers and don't have their own HTML pages, GitHub provides a fallback mechanism to generate a meta description tag for search engines.
> 
> GitHub uses the first few lines of the project's README file to generate the meta description tag. This is why you often see the first few lines of a project's README file appearing as the description in search engine results.
> 
> In addition to the README file, GitHub also uses other metadata from the project, such as the project's title and description from the repository settings, to generate the meta description tag. However, the README file is the primary source of information for the meta description tag.
> 
> 

>> What could be a possible reason as to why a totally false description is being displayed on the SERP's for the nicotine-plus project that includes unrelated content even though there are no mention of such content in the projects README file?

> There are several possible reasons why a false description might be displayed on the SERPs for the Nicotine+ project:
> 
> • Outdated or cached information: Search engines like Google cache web pages and their metadata, including meta descriptions, to improve performance. If the project's README file was updated recently, it's possible that the search engine is still displaying an outdated version of the description.
> • Scraped content from other sources: Search engines may scrape content from other sources, such as Wikipedia, online forums, or other websites, and use it to generate a description for the project. If the scraped content is incorrect or outdated, it may be displayed as the description.
> • Malicious activity: It's possible that someone has intentionally manipulated the search engine's index by creating fake or misleading content that is associated with the Nicotine+ project. This could be done through various means, such as creating fake web pages or submitting false metadata to the search engine.
> • Search engine algorithm error: Search engines use complex algorithms to generate descriptions, and sometimes these algorithms can make mistakes. It's possible that the algorithm has incorrectly generated a description for the Nicotine+ project.
> • Description from a fork or mirror: If there is a fork or mirror of the Nicotine+ project that has a different README file or metadata, it's possible that the search engine is displaying the description from the fork or mirror instead of the original project.
> • Description from a related project: If there is another project that is related to Nicotine+ and has a similar name or metadata, it's possible that the search engine is displaying the description from the related project instead of the Nicotine+ project.
> 
> It's worth noting that search engines like Google have mechanisms in place to prevent false or misleading information from being displayed, but sometimes errors can still occur.

>> Does GitHub use AI to generate content that appears in the descriptions for results in search engines?

> GitHub does not use AI to generate content that appears in the descriptions for results in search engines. As I mentioned earlier, GitHub uses the project's README file, title, and description from the repository settings to generate the meta description tag that is displayed in search engine results.
> 
> However, it's possible that the search engine itself, such as Google, may use AI-powered algorithms to generate or modify the description that is displayed in the search results. Google has confirmed that it uses AI-powered algorithms to generate snippets, which are the short summaries of a webpage's content that appear in search results.
> 
> Google's algorithms may analyze the content of the webpage, including the README file, and generate a snippet that it thinks best summarizes the content. However, this is done by the search engine, not by GitHub.
> 
> 

>> Does DuckDuckGo gather results from other search engines?

> DuckDuckGo is a search engine that aggregates results from multiple sources, including other search engines, but it does not directly gather results from other search engines in the classical sense.
> 
> Instead, DuckDuckGo uses a combination of its own web crawler, called DuckDuckBot, and partnerships with other search engines and data providers to gather its results. DuckDuckGo's search results are generated by combining the results from these various sources, including:
> 
> Its own web crawler, DuckDuckBot, which crawls and indexes web pages
> Bing, which provides a significant portion of DuckDuckGo's search results through a partnership
> Other search engines and data providers, such as Wikipedia, Wolfram Alpha, and more
> DuckDuckGo's algorithm then combines and ranks these results to provide a unique set of search results to the user.
> 
> 

>> Does DuckDuckGo use AI to generate description snippets for results displayed on the SERPs?

> DuckDuckGo does not use AI to generate description snippets for results displayed on the Search Engine Results Pages (SERPs). Instead, DuckDuckGo typically uses the meta description tag from the webpage itself, or if that's not available, it will use a snippet of text from the webpage that it thinks best summarizes the content.
> 
> DuckDuckGo's approach is to prioritize the original content and metadata provided by the webpage, rather than generating its own description using AI. This approach is in line with DuckDuckGo's focus on providing unbiased and unfiltered search results.

**Reply by slook** [reply]
> DuckDuckGo AI Chat BETA
> You are chatting with GPT-4o. AI chats may display inaccurate or offensive information.

>> What protocol does the nicotine-plus project use?

> The Nicotine-Plus project, which is an open-source client for the Soulseek file-sharing network, primarily uses the Soulseek protocol for its operations. Soulseek is a peer-to-peer file-sharing protocol that allows users to share and download music and other files. Nicotine-Plus enhances the original Soulseek client by providing additional features and a more user-friendly interface, while still adhering to the underlying protocol of Soulseek. If you need more specific details about the protocol or the project's features, feel free to ask!


>> What similarities are there between the Soulseek protocol and the BitTorrent protocol?

> The Soulseek protocol and the BitTorrent protocol are both used for peer-to-peer (P2P) file sharing, but they have different architectures and functionalities. Here are some similarities and differences between the two:
> 
> Similarities:
> 1. Peer-to-Peer Architecture: Both protocols allow users to share files directly with one another without the need for a central server to host the files.
> 2. Decentralization: Both systems are decentralized, meaning that files are distributed across multiple users' computers rather than being stored in a single location.
> 3. File Sharing: Both protocols enable users to upload and download files, allowing for the sharing of large amounts of data.
> 4. User Discovery: Both protocols have mechanisms for discovering other users and their shared files, although the methods of discovery differ.
>
> Differences:
> 1. File Distribution Method:
> - BitTorrent: Uses a "swarm" model where files are split into smaller pieces, and users can download pieces from multiple peers simultaneously. This increases download speeds and efficiency.
> - Soulseek: Typically shares whole files rather than breaking them into pieces. Users connect to specific peers to download entire files.
> 2. User Interface and Community:
> - Soulseek: Has a strong focus on music sharing and community, with features like chat rooms and user profiles that foster social interaction among users.
> - BitTorrent: More general-purpose and used for a wide variety of file types, not limited to music. It does not have the same community-oriented features as Soulseek.
> 3. Search Functionality:
> - Soulseek: Offers a built-in search feature that allows users to find specific files shared by other users in real-time.
> - BitTorrent: Typically relies on external torrent index sites to find .torrent files or magnet links, which contain metadata about the files being shared.
> 4. Protocol Complexity:
> - BitTorrent: Generally considered more complex due to its piece-based downloading, tracker systems, and support for various extensions (like DHT, PEX).
> - Soulseek: Simpler in terms of protocol design, focusing primarily on file sharing without the additional complexities of piece management.
>
> In summary, while both protocols facilitate peer-to-peer file sharing, they do so in different ways and cater to different user needs and communities.
> 

>> Wow, that's impressive knowledge. How did you discover that information?

> I have been trained on a diverse range of texts, including technical documentation, articles, and discussions about various technologies, including file-sharing protocols like Soulseek and BitTorrent. This training allows me to provide information and answer questions based on the knowledge I've acquired up until October 2023. If you have more questions or need further clarification on any topic, feel free to ask!


>> Does DuckDuckGo use any sort of AI to generate description snippets for projects like nicotine-plus, that are displayed in the SERPs?

> DuckDuckGo does not generate description snippets using AI for projects like Nicotine-Plus. Instead, it typically relies on information from various sources, including websites, user-contributed content, and structured data, to provide relevant snippets in the search engine results pages (SERPs). The goal is to present concise and accurate information based on the content available on the web. If you have specific questions about how DuckDuckGo operates or its features, feel free to ask!


>> Does GitHub use any sort of AI to generate description snippets for projects like nicotine-plus, that are displayed in the SERPs?

> GitHub does not specifically use AI to generate description snippets for projects displayed in search engine results pages (SERPs). Instead, the descriptions and metadata for GitHub repositories are typically provided by the repository owners themselves. When a project is created on GitHub, the owner can add a description, README file, and other relevant information that helps users understand the project.

> Search engines like Google or DuckDuckGo may then index this information and display it in their SERPs. While GitHub may use AI for various features within its platform, such as code suggestions or issue triaging, the snippets shown in search results are generally based on the content that users have provided in their repositories. If you have more questions about GitHub or related topics, feel free to ask!


>> What possible reasons are there that could explain why a description snippet on a DuckDuckGo SERP for a GitHub project contains content which is unrelated to the project?

> There are several possible reasons why a description snippet on a DuckDuckGo SERP for a GitHub project might contain content that is unrelated to the project:
> 
> 1. Indexing Errors: Search engines index content from various sources, and sometimes errors can occur during this process, leading to unrelated or outdated information being displayed.
> 2. Outdated Metadata: If the repository owner has changed the project description or README file but the search engine has not yet re-indexed the updated content, the snippet may show older, irrelevant information.
> 3. User-Contributed Content: If the project has a community aspect, such as discussions or issues, snippets may pull information from these areas, which could be unrelated to the main project description.
> 4. Similar Project Names: If there are multiple projects with similar names or keywords, the search engine might confuse them and display information from a different project.
> 5. Content from Forks or Related Repositories: If the project has been forked or is related to other repositories, the search engine might pull information from those related projects instead of the original.
> 6. SEO Manipulation: In some cases, repository owners might use SEO techniques that inadvertently lead to unrelated content being indexed, such as using misleading keywords or tags.
> 7. Third-Party Content: If the project has been discussed in articles, blogs, or forums, snippets from those sources might be indexed and displayed, leading to unrelated content appearing in the SERP.
> 8. Algorithm Limitations: Search engines use algorithms to determine what content to display, and these algorithms may not always accurately assess relevance, leading to unrelated snippets being shown.
> If you encounter such a situation, it may be helpful to check the actual GitHub repository for the most accurate and up-to-date information.

>> What does Soulseek server code 130 do?

> In the Soulseek protocol, server code 130 is used to indicate a "User is not logged in" error. This response is sent by the server when a user attempts to perform an action that requires them to be logged in, but they are not authenticated.
> 
> When a user receives this error, it typically means they need to log in to the Soulseek network before they can access certain features or perform specific actions, such as searching for files or connecting to other users.
> 
> If you have more questions about Soulseek or its protocol, feel free to ask!
> 

>> what about message 133?

> In the Soulseek protocol, server code 133 is used to indicate a "User is banned" error. This response is sent by the server when a user attempts to connect or perform an action but has been banned from the Soulseek network.
> 
> Being banned can occur for various reasons, such as violating the community guidelines or terms of service. When a user receives this error, it means they are not allowed to access the network or perform certain actions due to the ban.
> 
> If you have any more questions about Soulseek or related topics, feel free to ask!

>> Are you sure about that?

> I apologize for any confusion. In the Soulseek protocol, server code 133 actually indicates a "User is not found" error. This response is sent when a user attempts to perform an action related to another user (such as sending a message or trying to connect) but the specified user does not exist or is not currently online.
> 
> Thank you for your patience, and if you have any more questions or need further clarification, feel free to ask!
> 

>> Your answer is incorrect. Why could this be?

> I apologize for the inaccuracies in my previous responses regarding the Soulseek protocol codes. The errors could stem from a few factors:
> 
> 1. Limited Information: While I have access to a wide range of information, specific details about niche protocols like Soulseek may not be as thoroughly documented or may vary in interpretation across different sources.
> 2. Human Error: As an AI, I can make mistakes in recalling or interpreting information, especially if the context is not clear or if there are multiple interpretations of a term or code.
> 3. Updates and Changes: Protocols can evolve over time, and if there have been recent changes or updates to the Soulseek protocol that I am not aware of, that could lead to outdated or incorrect information.
> 4. If you have the correct information about server code 133 or any other specific details, I would appreciate your input. Thank you for your understanding, and I'm here to help with any further questions you may have!
> 

>> If input is provided to you, do you have the opportunity to include this in your next training dataset?

> No, I do not have the capability to learn from or incorporate individual user inputs into my training dataset. My responses are generated based on a fixed dataset that was last updated in October 2023, and I do not have the ability to update or modify that dataset based on interactions. Each session is stateless and does not retain memory of past interactions.
> 
> If you have feedback or corrections, they are valuable for understanding user needs, but they won't directly influence my responses in future interactions. If you have more questions or need assistance, feel free to ask!

>> What is server code 116?

> In the Soulseek protocol, server code 116 indicates a "User is already connected" error. This response is sent by the server when a user attempts to connect to another user who is already connected to them. Essentially, it means that the connection request cannot be fulfilled because the two users are already in a connected state.
> 
> If you have any more questions about Soulseek or its protocol, feel free to ask!


>> You have just made that up. It would be better to simply say that you don't have the information, rather than guessing.

> I apologize for the incorrect information and any confusion it caused. You're right; it would be better to acknowledge when I don't have specific details rather than providing inaccurate responses. Thank you for your feedback, and I appreciate your understanding. If you have any other questions or topics you'd like to discuss, please let me know!


>> No need to apologize. How do you try generate a guess when you arn't sure about something?

> When I generate responses, I rely on patterns and information from the training data I was built on. If I encounter a question where I don't have specific information, I may attempt to provide an answer based on related knowledge or context. However, this can sometimes lead to inaccuracies, especially in niche topics or specific technical details.
> 
> In cases where I'm uncertain, I should ideally indicate that I don't have the information rather than attempting to guess. Your feedback is valuable, and I appreciate your understanding as I strive to provide accurate and helpful responses. If you have more questions or need clarification on any topic, feel free to ask!

>> Is there any way to prevent you from making guesses during our interaction?

> While I don't have the ability to change my behavior or settings during our interaction, you can help guide the conversation by specifying that you prefer accurate information over guesses. If you ask a question and I don't have a clear answer, I can indicate uncertainty or acknowledge that I don't have the information rather than attempting to guess.
> 
> If you want to ensure that I provide only confirmed information, you can also ask for clarification or specify that you're looking for well-established facts. Your feedback and guidance during our conversation can help shape the responses you receive. If you have any specific preferences or questions, feel free to let me know!

>> I'd like to talk about where automatically generated description snippets on SERPs come from. You have already explained that missing data might lead to inaccurate information in AI responses, yet you have also indicated that search engines don't tend to use AI for generating content. So, there must be some other explaination then?

> Automatically generated description snippets on search engine results pages (SERPs) typically come from a variety of sources, rather than being generated by AI. Here are some common sources for these snippets:
> 
> 1. Meta Tags: Website owners can include meta description tags in their HTML. Search engines often use this content as the snippet if it is relevant to the search query.
> 2. Page Content: Search engines may extract text directly from the main content of a webpage, especially if it appears to be relevant to the user's query.
> 3. Structured Data: Websites that use structured data (like Schema.org markup) can provide specific information that search engines can use to generate rich snippets.
> 4. User-Contributed Content: For platforms like GitHub, snippets may come from user-generated content, such as project descriptions, README files, or discussions.
> 5. External Sources: Search engines may pull information from other websites that reference the page, such as reviews, articles, or social media mentions.
> 6. Algorithmic Selection: Search engines use algorithms to determine which content is most relevant to display based on the search query, which can lead to variations in snippets.
> 7. These methods can sometimes result in inaccuracies or outdated information if the source content has changed or if the search engine has not re-indexed the page.
> 

>> Can you make your response specific for projects on GitHub?

> For projects on GitHub, automatically generated description snippets on search engine results pages (SERPs) typically come from the following sources:
> 
> 1. Repository Description: The description provided by the repository owner in the GitHub settings is often used as the primary snippet. This is a brief summary that the owner can customize.
> 2. README File: If the repository has a README file, search engines may extract relevant content from it, as it often contains detailed information about the project, including its purpose, features, and usage instructions.
> 3. Issues and Discussions: Content from issues, pull requests, or discussions related to the project may also be indexed and used in snippets, especially if they contain relevant keywords or context.
> 4. User-Contributed Content: Comments or contributions from users can sometimes be indexed, particularly if they provide valuable insights or information about the project.
> 5. External References: Articles, blogs, or forums that discuss the GitHub project may also contribute to the snippets, especially if they link to the repository and provide context.
> 6. Structured Data: If the repository or associated documentation uses structured data markup, this can help search engines understand the content better and generate more accurate snippets.
> 7. These sources can lead to variations in the snippets displayed, and inaccuracies may arise if the repository information is outdated or if the search engine has not re-indexed the content.

>> Is all of the README file read by search engine crawlers? What about other markdown files in the project?

> Search engine crawlers typically index the content of README files in GitHub repositories, as these files are often the primary source of information about a project. However, whether all of the README file is read and indexed can depend on several factors, including:
> 
> 1. Content Length: Search engines may prioritize the most relevant sections of a README file, especially if it is lengthy. They might not index the entire file if they determine that only certain parts are pertinent to a user's query.
> 2. Relevance: Crawlers focus on content that is deemed relevant to search queries. If certain sections of the README or other markdown files do not contain relevant keywords or information, they may not be indexed.
> 3. Other Markdown Files: Other markdown files in the project (like CONTRIBUTING.md, CHANGELOG.md, etc.) can also be indexed, but their visibility in search results may vary. These files may not be as prominently featured as the README file, but they can still be crawled and indexed if they contain relevant content.
> 4. Search Engine Algorithms: Different search engines have varying algorithms that determine how they index content. Some may prioritize certain file types or structures over others.
> 
> In summary, while README files are a primary focus for indexing, other markdown files can also be indexed, but their prominence in search results may depend on their content and relevance to specific queries.

>> Do you think it might matter if the Soulseek protocol documentation is in a subfolder within the project, and if it isn't linked directly from the README file?

> Yes, the location of the Soulseek protocol documentation within a project and whether it is linked directly from the README file can significantly impact its visibility to search engine crawlers. Here are a few reasons why:
> 
> 1. Crawling Depth: Search engines typically prioritize content that is easily accessible from the main page (like the README file). If the documentation is in a subfolder and not linked from the README, it may be less likely to be crawled or indexed effectively.
> 2. Link Structure: Internal links help search engines understand the hierarchy and relationships between different files and folders in a project. If the README file does not link to the documentation, it may not be considered as important or relevant.
> 3. Relevance and Context: Content that is directly linked from the README file is often seen as more relevant to the project. If the documentation is not linked, it may not receive the same level of attention from search engines.
> 4. Indexing Frequency: Search engines may index content in subfolders less frequently than content that is more prominently featured. If the documentation is not easily discoverable, it might not be updated in search results as often.
> 
> In summary, having the Soulseek protocol documentation linked directly from the README file and ensuring it is easily accessible can improve its chances of being indexed and displayed in search engine results.

**Reply by slook** [reply]
This issue has now remedied itself:

![image](https://github.com/user-attachments/assets/3e6341e3-8f2a-406f-b569-cdb205d6d35f)

**Reply by mathiascode** [reply]
> Those hallucinated data are more likely to be coming from the `github.com` domain, since that is where many of the OpenAI (by Microsoft) models are being developed and have various sketchy implementations in CoPilot and whatnot. Alternatively, it may come from G or Bing, since DDG is not really an engine but is more like a search aggregator.

It's not only github.com. A lot of results on DuckDuckGo have generated descriptions, once you look closer at them:

![image](https://github.com/user-attachments/assets/8cd00beb-c6e7-455d-b5ed-1042c6ad6161)
![image](https://github.com/user-attachments/assets/c4b86906-b36e-4b21-b61c-724971b4c970)
![image](https://github.com/user-attachments/assets/d67bf0a4-9612-459f-a766-aab6bece6eaa)
![image](https://github.com/user-attachments/assets/9bad598c-dd32-4975-b67e-8b3939cc9ebd)

**Reply by slook** [reply]
https://www.theverge.com/2023/3/8/23629095/duckduckgo-ai-answers-browser-app-extension-search

https://arstechnica.com/information-technology/2023/03/wikipedia-ai-truth-duckduckgo-hopes-so-with-new-answerbot/

https://spreadprivacy.com/duckassist-launch/

> Update: July 2024. DuckAssist answers are now generated from sources beyond just Wikipedia. The answers are intentionally short — a starting point, rather than a deep dive — with prominently cited sources for you to click through and learn more. Not interested? DuckAssist is completely optional and easy to switch off in your [search settings](https://duckduckgo.com/settings?ref=spreadprivacy.com#aifeatures). For the most up-to-date information about this feature, please see the [DuckAssist Help Page](https://duckduckgo.com/duckduckgo-help-pages/results/duckassist/?ref=spreadprivacy.com).

> If you’re a publisher, you may opt out of being a potential source for DuckAssist answers by modifying the [robots.txt](https://metacpan.org/module/WWW::RobotRules) file for your domain(s). https://duckduckgo.com/duckduckgo-help-pages/results/duckassistbot/

----

> A lot of results on DuckDuckGo have generated descriptions

It does indeed look like the same concept as DuckAssist (which sometimes appears near the top of the serp as a header element) is now being integrated directly into some site descriptions. It doesn't seem to make any difference if DuckAssist is set to Never.

**Comment by slook** [comment]
**Incorrect download link to old version on Free Software Foundation Directory website**

The page at https://directory.fsf.org/wiki/Nicotine+ has a Download section containing a link to the 3.2.5 release on GitHub.

It was added by [Panos Alevropoulos](https://directory.fsf.org/wiki/User:Panos_Alevropoulos) who is an admin at FSD/FSF with a GNU mailbox and other contact details listed.

The page is editable by "users in the group: [Users](https://directory.fsf.org/wiki/Free_Software_Directory:Users)" with wiki form at link https://directory.fsf.org/wiki?title=Nicotine%2B&action=formedit

"create [a non-member account](https://my.fsf.org/user/register) without charge."

**Comment by slook** [comment]
**DeepWiki currently unclaimed auto-generated AI slop**

https://deepwiki.com/nicotine-plus/nicotine-plus

It contains many factual inaccuracies and irrelevant details about the project, which looks like it could be edited maybe it provide some kind of SEO benefit, it is listed in SERP for phase "pyNicotine" and maybe other terms. Nonetheless it is oddly interesting to read about how a robot perceives the codebase, and it made some nice flow charts such as...

<img width="1745" height="495" alt="image" src="https://github.com/user-attachments/assets/3e894088-5e5a-4d18-b310-a3009693f7ff" />

<img width="1743" height="858" alt="image" src="https://github.com/user-attachments/assets/cd8666e9-6ec3-486b-9dc2-d8ea64a7d016" />

<img width="1747" height="626" alt="image" src="https://github.com/user-attachments/assets/09090d27-82d1-46ad-a574-5f0308fcc263" />

<img width="1744" height="814" alt="image" src="https://github.com/user-attachments/assets/d8160a6c-41c9-474b-b7fa-5ecc8f64e48f" />

<img width="1742" height="902" alt="image" src="https://github.com/user-attachments/assets/3fd09ade-9f23-488a-b6c5-0489d90c6cc2" />

_Last indexed: 21 May 2025 ([ac7810](https://github.com/nicotine-plus/nicotine-plus/commits/ac7810b2))_

----

Update: The behaviour of this crawler can be customized with a `.devin/wiki.json` file see docs:
https://docs.devin.ai/work-with-devin/deepwiki#steering-deepwiki
https://cognition.ai/

**Comment by slook** [comment]
**DepScore - Code Analysis & Technical Insights**

https://depscore.com/posts/2025-10-03-nicotine-plus/

More AI slop. Doesn't appear to be editable.

_Analyzed: [61a17ed9](https://github.com/nicotine-plus/nicotine-plus/commit/61a17ed9fb196e91bdd575705261b7d015a4fe9c) from 2025-10-03_

**Comment by slook** [comment]
**Outdated hyperlink URL on Wikipedia page "List of file-sharing programs for Linux and BSD"**

https://en.wikipedia.org/wiki/List_of_file-sharing_programs_for_Linux_and_BSD#Soulseek

> ## [Soulseek](https://en.wikipedia.org/wiki/Soulseek)
> - [Nicotine Plus](https://en.wikipedia.org/wiki/Soulseek#Alternative_Clients) - [external] [Nicotine Plus](https://nicotine-plus.sourceforge.net/)

The external link is https://nicotine-plus.sourceforge.net/ which is an old version, whereas it should be the main project website at link https://nicotine-plus.org/ or the most relevant URL is https://nicotine-plus.org/doc/DOWNLOADS.html#gnulinux-bsd-haiku-illumos.

Also the anchor text should be "Nicotine+" rather than "Nicotine Plus".
